### PR TITLE
AssistantBuilder: Process Action take I

### PR DESCRIPTION
**Pull request by spolu**
## Description

Add base support for creating Process Action behind flag (disabled for everyone)

- Selection of dataSources
- Picking a TimeFrame

This is already a large PR so stopping for a first review here.

Screenshot from 2024-04-25 17-13-35

Next steps: 
- Ability so specify a schema in AssistantBuilder (pure UI after this one)
- Assistant to auto-generate the schema based on Instructions

## Risk

N/A

## Deploy Plan

- deploy `front`

### CODE REVIEW

File: front/components/assistant_builder/AssistantBuilder.tsx [hunk 1]
```diff
@@ -446,6 +447,21 @@ export default function AssistantBuilder({
       }
     }
 
+    if (builderState.actionMode === "PROCESS") {
```
Comment (PopDaph) [inline front/components/assistant_builder/AssistantBuilder.tsx:450]: Can you also edit line 282 (search on "// Not supported in the builder yet.") 🙏🏻

File: front/lib/api/assistant/configuration.ts [hunk 2]
```diff
@@ -1365,11 +1445,15 @@ async function _createAgentDataSourcesConfigData(
...
             parentsNotIn: dsConfig.filter.parents?.not,
             retrievalConfigurationId: retrievalConfigurationId,
+            processConfigurationId: processConfigurationId,
           },
           { transaction: t }
         );
       })
     );
+
+  console.log("HERE");
```
Comment (PopDaph) [inline front/lib/api/assistant/configuration.ts:1455]: There, and everywhere! 🎵

**Review by PopDaph — APPROVED** [review]
🚀

File: front/lib/api/assistant/configuration.ts (hunk 2 above)
Comment (spolu) [reply front/lib/api/assistant/configuration.ts:1455]: :pray:

File: front/components/assistant_builder/AssistantBuilder.tsx (hunk 1 above)
Comment (spolu) [reply front/components/assistant_builder/AssistantBuilder.tsx:450]: Yep adding "PROCESS" there even if no impact for now

**Pull request merged**